gh-97661: Improve accuracy of sqlite3.Cursor.fetchone docs (#97662)
python/cpython: Doc/library/sqlite3.rst

   .. method:: fetchone()

      Return the next row of a query result set as a :class:`tuple` or an object

>       returned by :attr:`~Connection.row_factory`.

Reviewer: The actual behaviour is (pseudo code):

```python
# simplified version of the sqlite3.Cursor `tp_iternext` function (AKA the __next__ method)
# self is a pysqlite_Cursor struct
def __next__(self):
    row = fetch_one_row(self)  # returns row as a tuple
    if self.row_factory is not None:
        return self.row_factory(self, row)
    return row
```

And in `fetchone()` (pseudo code):

```python
# self is a sqlite3.Cursor object
def fetchone(self):
    return next(self)  # calls tp_iternext (aka __next__)
```

Reviewer: The reference should explain this as concise as possible. I think it is worth it to point out that `row_factory` is only used if it is not `None`. The wording quickly becomes too clumsy for a reference, IMO:

_If row_factory is not None, return the next row of a query result set as a tuple. Else ..._

Perhaps using more explicit Sphinx markup will help:

```suggestion
      :return: The next row of a query result set.
      :rtype: None, tuple, ...
```

@CAM-Gerlach, what do you think?

Reviewer: Using multiple `:rtype:` directives quickly becomes ugly as well, so might be better to keep it as prose.

Reviewer 2: For now, what about

```suggestion
      If :attr:`~Connection.row_factory` is ``None``,
      return the next row query result set as a :class:`tuple`.
      Else, pass it to the row factory and return its result.
```
